.github/workflows/binary-installers.yml

```diff
@@ -0,0 +1,64 @@
+name: Create binary installers
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  pip-compile:
+    strategy:
+      fail-fast: false
+      matrix:
+        output-file:
+          - py3.10-darwin-x86_64-reqs
+          - py3.10-linux-x86_64-cuda-reqs
+          - py3.10-windows-x86_64-cuda-reqs
+        python-version:
+          - '3.9'
+          - '3.10'
+        include:
+          - output-file: py3.10-darwin-x86_64-reqs
+            os: macos-12
+            requirements-file: requirements-mac-mps-cpu.txt
+          - output-file: py3.10-linux-x86_64-cuda-reqs
+            os: ubuntu-22.04
+            requirements-file: requirements-lin-cuda.txt
+          - output-file: py3.10-windows-x86_64-cuda-reqs
+            os: windows-2022
+            requirements-file: requirements-win-colab-cuda.txt
+    name: ${{ matrix.output-file }} on Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout sources
+        id: checkout-sources
+        uses: actions/checkout@v3
+
+      - name: setup python
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: install pip-tools
+        id: install-pip-tools
+        run: pip3 install pip-tools
+
+      - name: debug output
+        run: |
+          pip3 -V
+          python3 -V
+          pip3 list
+
+      - name: run pip-compile
+        id: pip-compile
+        run: pip-compile --allow-unsafe --generate-hashes --output-file=binary_installer/${{ matrix.output-file }}.txt binary_installer/requirements.in
```

Comment:
This is not something which should be run frequently. Dependent package versions can change *daily*. We should only be recompiling from scratch if

1) a dependent package *must* be updated (it's Corrupting data, there's a security issue, etc.)
2) (possibly) something *we* do makes us incompatible with a specific package version and we need to move to a compatible one
3) we *add* a dependent package (and even in that case this command is incorrect: the correct command includes `reuse-hashes`)

*Removing* a dependent package likewise only requires deleting the dependency from requirements.in and the various requirements files (no pip-compile is required at all)

Either way, the full command for generating ALL dependencies is `pip-compile -r --annotate --annotation-style split --generate-hashes --emit-index-url --allow-unsafe --resolver=backtracking --output-file=binary_installer/${{ matrix.output-file }}.txt binary_installer/requirements.in`; the full command for UPDATING is `pip-compile -r --annotate --annotation-style split --generate-hashes --emit-index-url --allow-unsafe --resolver=backtracking --reuse-hashes --output-file=binary_installer/${{ matrix.output-file }}.txt binary_installer/requirements.in` (note that the only change is *adding* `--reuse-hashes`)

Reply:
@tildebyte so should I use the `generate all deps` or the `update deps` command? If I understand you right, point 3 mentions to use the `update deps` version, right?


Problem for Windows generation of reequirements currently is

```
There are incompatible versions in the resolved dependencies:
  basicsr==1.4.1 (from gfpgan==1.3.8->-r binary_installer/requirements.in (line 29))
  basicsr>=1.4.2 (from realesrgan==0.3.0->-r binary_installer/requirements.in (line 15))
```

which of course is also not fixed by neither one or the other of the commands you mentioned (I just tested this in my fork now).

Reply:
> BTW: My plan is to run this command only when changes to the requirements.in file have been done (which would be the only reason to update the requirement files)

That's perfect